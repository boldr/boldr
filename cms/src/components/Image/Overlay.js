import React, { Component, PropTypes } from 'react';
import ReactDOM from 'react-dom';

class Overlay extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isVisible: false
    };
  }

  componentDidMount() {
    this.setState({ isVisible: true }); // eslint-disable-line
  }

  componentWillReceiveProps(nextProps) {
    if (!nextProps.isVisible) this.setState({ isVisible: false });
  }

  shouldComponentUpdate(nextProps, nextState) {
    return this.props.isVisible !== nextProps.isVisible ||
      this.state.isVisible !== nextProps.isVisible;
  }

  getStyle() {
    const opacity = this.state.isVisible & 1; // bitwise and; converts bool to 0 or 1
    return Object.assign({}, defaults.styles.overlay, { opacity });
  }
  render() {
    return <div style={ this.getStyle() } />;
  }
}

Overlay.propTypes = {
  isVisible: bool.isRequired
};
export default Overlay;
