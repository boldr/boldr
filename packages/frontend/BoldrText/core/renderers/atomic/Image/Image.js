import React from 'react';
import { List } from 'immutable';
import { EditorState } from 'draft-js';
import { setBlockData } from '../../../../utils/block';
import { selectBlock, removeBlock } from '../../../../utils/editor';
import Switch from '../../../../components/Switch';

export default class Image extends React.Component {
  state = {
    toolbarVisbile: false,
    toolbarOffset: 0,
    linkEditorVisible: false,
  };

  render() {
    const { mediaData, language } = this.props;
    const { toolbarVisbile, toolbarOffset, linkEditorVisible } = this.state;
    const blockData = this.props.block.getData();

    const float = blockData.get('float');
    let alignment = blockData.get('alignment');
    const { url, link, linkTarget, width, height } = mediaData;
    const imageStyles = {};
    let clearFix = false;

    if (float) {
      alignment = null;
    } else if (alignment === 'left') {
      imageStyles.float = 'left';
      clearFix = true;
    } else if (alignment === 'right') {
      imageStyles.float = 'right';
      clearFix = true;
    } else if (alignment === 'center') {
      imageStyles.textAlign = 'center';
    } else {
      imageStyles.float = 'left';
      clearFix = true;
    }

    return (
      <div className="be-media__embeder">
        <div
          style={imageStyles}
          className="be-embed__image"
          onMouseOver={this.showToolbar}
          onMouseLeave={this.hideToolbar}
        >
          {toolbarVisbile &&
            <div
              style={{ marginLeft: toolbarOffset }}
              ref={instance => (this.toolbarElement = instance)}
              data-float={float}
              data-alignment={alignment}
              className="be-embed__image-toolbar"
            >
              {linkEditorVisible &&
                <div onClick={this.preventDefault} className="be-embed__image-link-editor">
                  <input
                    type="text"
                    placeholder={language.linkEditor.inputPlaceHolder}
                    onKeyDown={this.setImageLink}
                    defaultValue={link}
                  />
                  <div className="be-switch__group">
                    <Switch
                      active={linkTarget === '_blank'}
                      onClick={() => this.setImageLinkTarget(linkTarget)}
                    />
                    <label>
                      {language.linkEditor.openInNewWindow}
                    </label>
                  </div>
                </div>}
              <a data-float="left" onClick={this.setImageFloat}>
                &#xe91e;
              </a>
              <a data-float="right" onClick={this.setImageFloat}>
                &#xe914;
              </a>
              <a data-alignment="left" onClick={this.setImageAlignment}>
                &#xe027;
              </a>
              <a data-alignment="center" onClick={this.setImageAlignment}>
                &#xe028;
              </a>
              <a data-alignment="right" onClick={this.setImageAlignment}>
                &#xe029;
              </a>
              <a className={link ? 'active' : ''} onClick={this.toggleLinkEditor}>
                &#xe91a;
              </a>
              <a onClick={this.removeImage}>&#xe9ac;</a>
              <i
                style={{ marginLeft: toolbarOffset * -1 }}
                className="be-embed__image-toolbar-arrow"
              />
            </div>}
          <img
            ref={instance => (this.imageElement = instance)}
            src={url}
            width={width}
            height={height}
          />
        </div>
        {clearFix && <div className="clearfix" style={{ clear: 'both' }} />}
      </div>
    );
  }

  calcToolbarOffset() {
    const viewWidth = document.body.getBoundingClientRect().width;
    const toolbarRect = this.toolbarElement.getBoundingClientRect();
    const imageRect = this.imageElement.getBoundingClientRect();
    let right = imageRect.right - imageRect.width / 2 + toolbarRect.width / 2;
    const left = imageRect.left + imageRect.width / 2 - toolbarRect.width / 2;

    right = viewWidth - right;

    if (right < 10) {
      return right - 10;
    } else if (left < 10) {
      return left * -1 + 10;
    } else {
      return 0;
    }
  }

  removeImage = e => {
    const { block, getEditorState, onChange } = this.props;
    onChange(removeBlock(getEditorState(), block));
    this.props.setEditorProp('readOnly', false);
  };

  toggleLinkEditor = () => {
    this.setState({
      linkEditorVisible: !this.state.linkEditorVisible,
    });
  };

  setImageLink = e => {
    if (e.keyCode !== 13) {
      return;
    }

    const link = e.currentTarget.value.trim();
    const { entityKey, contentState, editorState, onChange } = this.props;
    contentState.mergeEntityData(entityKey, { link });

    onChange(EditorState.push(editorState, contentState, 'change-block-data'));
    setImmediate(this.props.forceRender);
  };

  setImageLinkTarget(linkTarget) {
    const { entityKey, contentState, editorState, onChange } = this.props;
    linkTarget = linkTarget === '_blank' ? '' : '_blank';
    contentState.mergeEntityData(entityKey, { linkTarget });

    onChange(EditorState.push(editorState, contentState, 'change-block-data'));
    setImmediate(this.props.forceRender);
  }

  setImageFloat = e => {
    let { float } = e.currentTarget.dataset;
    const { block, getEditorState, contentState, onChange } = this.props;
    const blockData = block.getData();
    const lastFloat = blockData.get('float');

    if (lastFloat === float) {
      float = null;
    }

    onChange(setBlockData(selectBlock(getEditorState(), block), { float }));
    this.props.setEditorProp('readOnly', false);
  };

  setImageAlignment = e => {
    let { alignment } = e.currentTarget.dataset;
    const { block, getEditorState, contentState, onChange } = this.props;
    const blockData = block.getData();
    const lastAlignment = blockData.get('alignment');

    if (lastAlignment === alignment) {
      alignment = null;
    }

    onChange(setBlockData(selectBlock(getEditorState(), block), { alignment }));
    this.props.setEditorProp('readOnly', false);
  };

  showToolbar = () => {
    if (!this.state.toolbarVisbile) {
      this.setState(
        {
          toolbarVisbile: true,
        },
        () => {
          this.props.setEditorProp('readOnly', true);
          this.setState({
            toolbarOffset: this.calcToolbarOffset(),
          });
        },
      );
    }
  };

  hideToolbar = () => {
    this.setState(
      {
        toolbarVisbile: false,
      },
      () => {
        this.props.setEditorProp('readOnly', false);
      },
    );
  };
}
