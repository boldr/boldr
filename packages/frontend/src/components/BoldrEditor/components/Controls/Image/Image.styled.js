/* eslint-disable no-unused-vars */
import styled from 'styled-components';

export const ImageModal = styled.div`
  position: absolute;
  top: 35px;
  left: 5px;
  z-index: 100;
  display: flex;
  flex-direction: column;
  width: 235px;
  color: #f1f1f1;
  height: 200px;
  padding: 15px;
  background-color: #243140;
  border-radius: 3px;
  box-shadow: 2px 10px 30px rgba(#000, 0.3);
`;

export const ImageHeader = styled.div`
  display: flex;
  margin: 10px 0;
  font-size: 15px;
`;

export const ImageOption = styled.span`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 50%;
  cursor: pointer;
`;

export const ImageLabel = styled.span`
  width: 80px;
  background: ${props => (props.highlighted ? '#6eb8d4' : '#f1f1f1')};
  border: 1px solid #f1f1f1;
  border-bottom: ${props => (props.highlighted ? '2px solid #0a66b7' : 'inherit')};
  margin-top: 5px;
`;

export const UploadOpt = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 65px;
  color: gray;
  background-color: #f1f1f1;
  border: none;
  margin: 10px 0;
  outline: ${props => (props.highlighted ? '2px dashed #0a66b7' : ' 2px dashed #eaeaea')};
  outline-offset: -10px;
  font-size: 15px;
  cursor: pointer;
`;

export const UploadLabel = styled.label`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  padding: 15px;
  cursor: pointer;

  span {
    padding: 0 20px;
  }
`;

export const UploadInput = styled.input`
  position: absolute;
  z-index: -1;
  overflow: hidden;
  width: 0.1px;
  height: 0.1px;
  opacity: 0;
`;
export const UrlSection = styled.div`display: block;`;
export const UploadUrlInput = styled.input`
  width: 95%;
  height: 35px;
  padding: 0 5px;
  border: 1px solid #f1f1f1;
  border-radius: 2px;
  margin: 30px 0 20px;
  font-size: 15px;
`;
export const SizeSection = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 8px 0;
`;
export const SizeInput = styled.input`
  width: 45%;
  height: 20px;
  border: 1px solid #f1f1f1;
  border-radius: 2px;
  font-size: 12px;
`;

export const ImageBtnSection = styled.div`margin: 10px auto 0;`;

export const ImageBtn = styled.button`
  width: 75px;
  height: 30px;
  background: white;
  border: 1px solid #f1f1f1;
  border-radius: 2px;
  margin: 0 5px;
  text-transform: capitalize;
  cursor: pointer;

  &:hover {
    box-shadow: 1px 1px 0px #bfbdbd;
  }

  &:active {
    box-shadow: 1px 1px 0px #bfbdbd inset;
  }

  &:focus {
    outline: none !important;
  }

  &:disabled {
    background: #ece9e9;
  }
`;
