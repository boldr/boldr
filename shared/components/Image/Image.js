/* @flow */
import React from 'react';

type Props = {
  className: String,
  alt: String,
  width: String,
  imgSrc: String,
};

const Image = (props: Props) => {
  const imgClass = props.className ? props.className : 'boldr-image';
  return (
    <img src={ props.imgSrc } className={ imgClass } alt={ props.alt } width={ props.width } />
  );
};

export default Image;
