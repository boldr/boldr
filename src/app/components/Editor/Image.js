import React, { Component } from 'react';

const styles = {
  image: {
    maxWidth: '100%'
  }
};

export default class Image extends React.Component {
  render() {
    return (
      <div>
        <img contentEditable={ false }
          src={ this.props.src }
          style={ Object.assign({}, styles.image, this.props.style) }
        />
      </div>
    );
  }
}
