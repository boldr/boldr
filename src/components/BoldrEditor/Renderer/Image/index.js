/* @flow */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { EditorState } from 'draft-js';
import classNames from 'classnames';
import Option from '../../components/Option';

const getImageComponent = config =>
  class Image extends Component {
    static propTypes: Object = {
      block: PropTypes.object,
      contentState: PropTypes.object,
    };

    state: Object = {
      hovered: false,
    };

    setEntityAlignmentLeft: Function = (): void => {
      this.setEntityAlignment('left');
    };

    setEntityAlignmentRight: Function = (): void => {
      this.setEntityAlignment('right');
    };

    setEntityAlignmentCenter: Function = (): void => {
      this.setEntityAlignment('none');
    };

    setEntityAlignment: Function = (alignment): void => {
      const { block, contentState } = this.props;
      const entityKey = block.getEntityAt(0);
      contentState.mergeEntityData(entityKey, { alignment });
      config.onChange(EditorState.push(config.getEditorState(), contentState, 'change-block-data'));
      this.setState({
        dummy: true,
      });
    };

    toggleHovered: Function = (): void => {
      const hovered = !this.state.hovered;
      this.setState({
        hovered,
      });
    };

    renderAlignmentOptions(alignment): Object {
      return (
        <div
          className={classNames('boldr-editor__image-alignment-options-popup', {
            'boldr-editor__image-alignment-options-popup-right': alignment === 'right',
          })}
        >
          <Option
            onClick={this.setEntityAlignmentLeft}
            className="boldr-editor__image-alignment-option"
          >
            L
          </Option>
          <Option
            onClick={this.setEntityAlignmentCenter}
            className="boldr-editor__image-alignment-option"
          >
            C
          </Option>
          <Option
            onClick={this.setEntityAlignmentRight}
            className="boldr-editor__image-alignment-option"
          >
            R
          </Option>
        </div>
      );
    }

    render(): Object {
      const { block, contentState } = this.props;
      const { hovered } = this.state;
      const { isReadOnly, isImageAlignmentEnabled } = config;
      const entity = contentState.getEntity(block.getEntityAt(0));
      const { src, alignment, height, width } = entity.getData();

      return (
        <span
          onMouseEnter={this.toggleHovered}
          onMouseLeave={this.toggleHovered}
          className={classNames('boldr-editor__image-alignment', {
            'boldr-editor__image-left': alignment === 'left',
            'boldr-editor__image-right': alignment === 'right',
            'boldr-editor__image-center': !alignment || alignment === 'none',
          })}
        >
          <span className="boldr-editor__image-imagewrapper">
            <img
              src={src}
              alt=""
              style={{
                height,
                width,
              }}
            />
            {!isReadOnly() && hovered && isImageAlignmentEnabled()
              ? this.renderAlignmentOptions(alignment)
              : undefined}
          </span>
        </span>
      );
    }
  };

export default getImageComponent;
