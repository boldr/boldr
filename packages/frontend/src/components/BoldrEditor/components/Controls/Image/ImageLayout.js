/* eslint-disable no-continue */
/* @flow */

import * as React from 'react';
import classNames from 'classnames';
import Image from '@boldr/icons/Image';
import Option from '../../Option';
import Spinner from '../../Spinner';
import { ControlWrapper } from '../Controls.styled';
import type { ImageConfig } from '../../../core/config';
import {
  ImageModal,
  ImageHeader,
  ImageOption,
  ImageLabel,
  UploadOpt,
  UploadInput,
  UploadUrlInput,
  UrlSection,
  UploadLabel,
  SizeSection,
  SizeInput,
  ImageBtn,
  ImageBtnSection,
} from './Image.styled';

export type Props = {
  expanded?: boolean,
  doCollapse: Function,
  onExpandEvent?: Function,
  config: ImageConfig,
  onChange?: Function,
};

type State = {
  imgSrc: string,
  dragEnter: boolean,
  uploadHighlighted: boolean,
  showImageLoading: boolean,
  height: 'auto' | number,
  width: 'auto' | number,
  alt: string,
};

class ImageLayout extends React.Component<Props, State> {
  state: State = {
    imgSrc: '',
    dragEnter: false,
    uploadHighlighted: this.props.config.uploadEnabled && Boolean(this.props.config.uploadCallback),
    showImageLoading: false,
    height: this.props.config.defaultSize.height,
    width: this.props.config.defaultSize.width,
  };

  componentWillReceiveProps(props: Props): void {
    if (this.props.expanded && !props.expanded) {
      this.setState({
        imgSrc: '',
        dragEnter: false,
        uploadHighlighted:
          this.props.config.uploadEnabled && Boolean(this.props.config.uploadCallback),
        showImageLoading: false,
        height: this.props.config.defaultSize.height,
        width: this.props.config.defaultSize.width,
      });
    } else if (
      props.config.uploadCallback !== this.props.config.uploadCallback ||
      props.config.uploadEnabled !== this.props.config.uploadEnabled
    ) {
      this.setState({
        uploadHighlighted: props.config.uploadEnabled && Boolean(props.config.uploadCallback),
      });
    }
  }

  props: Props;

  updateValue: Function = (event: Object): void => {
    this.setState({
      [`${event.target.name}`]: event.target.value,
    });
  };

  toggleShowImageLoading: Function = (): void => {
    const showImageLoading = !this.state.showImageLoading;
    this.setState({
      showImageLoading,
    });
  };

  showImageURLOption: Function = (): void => {
    this.setState({
      uploadHighlighted: false,
    });
  };

  showImageUploadOption: Function = (): void => {
    this.setState({
      uploadHighlighted: true,
    });
  };

  addImageFromState: Function = (): void => {
    const { imgSrc, height, width } = this.state;
    const { onChange } = this.props;
    onChange(imgSrc, height, width);
  };

  addImageFromSrcLink: Function = (imgSrc: string): void => {
    const { height, width } = this.state;
    const { onChange } = this.props;
    onChange(imgSrc, height, width);
  };

  onImageDrop: Function = (event: Object): void => {
    event.preventDefault();
    event.stopPropagation();
    this.setState({
      dragEnter: false,
    });

    // Check if property name is files or items
    let data = event.dataTransfer.items;
    let dataIsItems = true;

    // IE uses 'files' instead of 'items'
    if (!data) {
      data = event.dataTransfer.files;
      dataIsItems = false;
    }

    for (let i = 0; i < data.length; i += 1) {
      if (data[i].kind === 'string' && data[i].type.match('^text/plain')) {
        // This item is the target node
        continue;
      } else if (data[i].kind === 'string' && data[i].type.match('^text/html')) {
        // Drag data item is HTML
        continue;
      } else if (data[i].kind === 'string' && data[i].type.match('^text/uri-list')) {
        // Drag data item is URI
        continue;
      } else if ((!dataIsItems || data[i].kind === 'file') && data[i].type.match('^image/')) {
        // Drag data item is an image file
        const file = dataIsItems ? data[i].getAsFile() : data[i];
        this.uploadImage(file);
      }
    }
  };

  onDragEnter: Function = (event: SyntheticEvent<>): void => {
    this.stopPropagation(event);
    this.setState({
      dragEnter: true,
    });
  };

  selectImage: Function = (event: Object): void => {
    if (event.target.files && event.target.files.length > 0) {
      this.uploadImage(event.target.files[0]);
    }
  };

  uploadImage: Function = (file: Object): void => {
    this.toggleShowImageLoading();
    const { uploadCallback, fileUrl } = this.props.config;
    uploadCallback(file)
      .then(({ data }) => {
        this.setState({
          showImageLoading: false,
          dragEnter: false,
        });
        // @note: data.uploadMedia is the GraphQL mutation.
        return this.addImageFromSrcLink(`${fileUrl}/${data.uploadMedia.name}`);
      })
      .catch(() => {
        this.setState({
          showImageLoading: false,
          dragEnter: false,
        });
      });
  };

  fileUploadClick = event => {
    this.fileUpload = true;
    event.stopPropagation();
  };

  stopPropagation: Function = (event: Object): void => {
    if (!this.fileUpload) {
      event.preventDefault();
      event.stopPropagation();
    } else {
      this.fileUpload = false;
    }
  };

  renderAddImageModal(): React.Node {
    const { imgSrc, uploadHighlighted, showImageLoading, dragEnter, height, width } = this.state;
    const {
      config: { modalClassName, uploadCallback, uploadEnabled, urlEnabled, inputAccept },
    } = this.props;
    return (
      <ImageModal className={modalClassName} onClick={this.stopPropagation}>
        <ImageHeader>
          {uploadEnabled &&
            uploadCallback &&
            <ImageOption onClick={this.showImageUploadOption}>
              Upload
              <ImageLabel highlighted={uploadHighlighted} />
            </ImageOption>}
          {urlEnabled &&
            <ImageOption onClick={this.showImageURLOption}>
              Upload URL
              <ImageLabel highlighted={!uploadHighlighted} />
            </ImageOption>}
        </ImageHeader>
        {uploadHighlighted
          ? <div onClick={this.fileUploadClick}>
              <UploadOpt
                onDragEnter={this.onDragEnter}
                onDragOver={this.stopPropagation}
                onDrop={this.onImageDrop}
                highlighted={dragEnter}>
                <UploadLabel htmlFor="file">Drop the file or click to upload</UploadLabel>
              </UploadOpt>
              <UploadInput type="file" id="file" accept={inputAccept} onChange={this.selectImage} />
            </div>
          : <UrlSection>
              <UploadUrlInput
                placeholder="Enter url"
                name="imgSrc"
                onChange={this.updateValue}
                onBlur={this.updateValue}
                value={imgSrc}
              />
            </UrlSection>}
        <SizeSection>
          ↕&nbsp;
          <SizeInput
            onChange={this.updateValue}
            onBlur={this.updateValue}
            value={height}
            name="height"
            placeholder="Height"
          />
          &nbsp;↔&nbsp;
          <SizeInput
            onChange={this.updateValue}
            onBlur={this.updateValue}
            value={width}
            name="width"
            placeholder="Width"
          />
        </SizeSection>
        <ImageBtnSection>
          <ImageBtn onClick={this.addImageFromState} disabled={!imgSrc || !height || !width}>
            Add
          </ImageBtn>
          <ImageBtn onClick={this.props.doCollapse}>Cancel</ImageBtn>
        </ImageBtnSection>
        {showImageLoading
          ? <div className="be-image__modal-spinner">
              <Spinner />
            </div>
          : undefined}
      </ImageModal>
    );
  }

  render(): React.Node {
    const { config: { className, title }, expanded } = this.props;
    return (
      <ControlWrapper aria-haspopup="true" aria-expanded={expanded} aria-label="be-image__control">
        <Option
          className={classNames(className)}
          value="unordered-list-item"
          onClick={this.props.onExpandEvent}
          title={title}>
          <Image color="#222" />
        </Option>
        {expanded ? this.renderAddImageModal() : undefined}
      </ControlWrapper>
    );
  }
}

export default ImageLayout;
