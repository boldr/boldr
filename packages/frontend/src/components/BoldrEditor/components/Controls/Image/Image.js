/* @flow */

import React, { Component } from 'react';
// $FlowIssue
import { AtomicBlockUtils } from 'draft-js';
import ImageLayout from './ImageLayout';

export type Props = {
  onChange: Function,
  editorState: Object,
  config: Object,
};
class Image extends Component {
  state: Object = {
    expanded: false,
  };

  props: Props;

  expandCollapse: Function = (): void => {
    this.setState({
      expanded: this.signalExpanded,
    });
    this.signalExpanded = false;
  };

  handleExpandEvent: Function = (): void => {
    this.signalExpanded = !this.state.expanded;
  };

  handleExpand: Function = (): void => {
    this.setState({
      expanded: true,
    });
  };

  handleCollapse: Function = (): void => {
    this.setState({
      expanded: false,
    });
  };

  addImage: Function = (src: string, height: string, width: string, alt: string): void => {
    const { editorState, onChange, config } = this.props;
    const entityData = { src, height, width };
    if (config.alt.present) {
      entityData.alt = alt;
    }
    const entityKey = editorState
      .getCurrentContent()
      .createEntity('IMAGE', 'MUTABLE', { src, height, width })
      .getLastCreatedEntityKey();
    const newEditorState = AtomicBlockUtils.insertAtomicBlock(editorState, entityKey, ' ');
    onChange(newEditorState);
    this.doCollapse();
  };

  render(): Object {
    const { config } = this.props;
    const { expanded } = this.state;
    const ImageComponent = ImageLayout;
    return (
      <ImageComponent
        config={config}
        onChange={this.addImage}
        expanded={expanded}
        onExpandEvent={this.handleExpandEvent}
        doExpand={this.handleExpand}
        doCollpase={this.handleCollapse}
      />
    );
  }
}

export default Image;
