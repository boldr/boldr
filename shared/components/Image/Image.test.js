import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Image from './Image';

it('<Image /> renders the Image svg component', () => {
  const wrapper = shallow(<Image />);
  const wrapperClass = wrapper.find('.boldr-image');
  expect(wrapperClass.is('.boldr-image')).toBe(true);
});

it('renders snapshot', () => {
  const wrapper = shallow(<Image />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
