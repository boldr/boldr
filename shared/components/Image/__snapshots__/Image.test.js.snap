exports[`test renders snapshot 1`] = `
<img
  className="boldr-image" />
`;
